### PR TITLE
Build system: Build pgxn zip file with 'make pgxn'

### PR DESCRIPTION
**Pull request by fschopp**
Jira: MADLIB-623

Added a target for PGXN to the build system. The release number is set in
deploy/PGXN/CMakeLists.txt, the version number is taken from
src/config/Version.yml. This packages the current source repository, so it is best
used on a clean repository.

### CODE REVIEW

File: deploy/PGXN/generate_package.sh.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+TEMPDIR=`mktemp -d -t madlib`
+"@CMAKE_COMMAND@" -E create_symlink \
+    "@CMAKE_SOURCE_DIR@" \
+    "${TEMPDIR}/@MADLIB_PGXN_VERSION_STR@"
+"@CMAKE_COMMAND@" -E create_symlink \
+    "@CMAKE_CURRENT_BINARY_DIR@/zipignore" \
+    "${TEMPDIR}/zipignore"
+cd "${TEMPDIR}"
+zip --exclude @zipignore \
+    -r "@CMAKE_CURRENT_BINARY_DIR@/madlib.zip" \
```
Comment (fschopp) [inline deploy/PGXN/generate_package.sh.in]: Per our discussion, the filename should be more descriptive, maybe `madlib-pgxn-@MADLIB_PGXN_VERSION_STRING@.zip`. @haradh1 Do you agree?

File: deploy/PGXN/generate_package.sh.in [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "@MADLIB_PGXN_VERSION_STR@"
```
Comment (fschopp) [inline deploy/PGXN/generate_package.sh.in]: Likewise, the top level directory in the zip file should have the same name (without the .zip suffix) as the zip file, right?

File: deploy/PGXN/zipignore.in [hunk 3]
```diff
@@ -0,0 +1,3 @@
+*/build*
+*/.DS_Store
+*/.git/*
```
Comment (fschopp) [inline deploy/PGXN/zipignore.in]: The ideal solution would be to populate this from the .gitignore file(s). For now, maybe I should just add those entries manually.

File: deploy/PGXN/META.json.in [hunk 4]
```diff
@@ -7,9 +7,9 @@
```
Comment (haradh1) [inline deploy/PGXN/META.json.in:1]: PGXN_VERSION should also be configured.

File: deploy/PGXN/generate_package.sh.in (hunk 1 above)
Comment (haradh1) [inline deploy/PGXN/generate_package.sh.in]: Yes I do.

File: deploy/PGXN/META.json.in [hunk 5]
```diff
@@ -7,9 +7,9 @@
     "license": "bsd",
     "provides": {
         "madlib": {
-            "file": "madlib--0.4.0.sql",
```
Comment (haradh1) [inline deploy/PGXN/META.json.in:10]: Hmm, this should match .control file, which is in src/ports/postgres/extension

File: deploy/PGXN/generate_package.sh.in (hunk 2 above)
Comment (haradh1) [inline deploy/PGXN/generate_package.sh.in]: It is not necessarily the same,  but it looks any directory name works fine.

File: deploy/PGXN/META.json.in (hunk 5 above)
Comment (fschopp) [inline deploy/PGXN/META.json.in:10]: So would it be OK, to replace

```
default_version = '@MADLIB_VERSION_MAJOR@.@MADLIB_VERSION_MINOR@.@MADLIB_VERSION_PATCH@'
```

by

```
default_version = '@MADLIB_VERSION_STRING@'
```

in that file?

File: deploy/PGXN/META.json.in (hunk 4 above)
Comment (fschopp) [inline deploy/PGXN/META.json.in:1]: Done.

File: deploy/PGXN/zipignore.in (hunk 3 above)
Comment (fschopp) [inline deploy/PGXN/zipignore.in]: Done.

File: deploy/PGXN/META.json.in (hunk 5 above)
Comment (haradh1) [inline deploy/PGXN/META.json.in:10]: If we change control file, we should change madlib--{version}.sql file respectively.

Comment (fschopp) [inline deploy/PGXN/META.json.in:10]: Hmm, madlib--{version}.sql is a generated file. So I do not see any more why the .control file should be changed.

Comment (haradh1) [inline deploy/PGXN/META.json.in:10]: What I mean is we have to keep the same name in control file as sql file

File: deploy/PGXN/META.json.in [hunk 6]
```diff
@@ -2,14 +2,14 @@
...
+    "version": "@MADLIB_PGXN_VERSION_STR@",
     "maintainer": "MADlib development team",
     "license": "bsd",
     "provides": {
         "madlib": {
-            "file": "madlib--0.4.0.sql",
+            "file": "madlib--${MADLIB_VERSION_MAJOR}.${MADLIB_VERSION_MINOR}.${MADLIB_VERSION_PATCH}.sql",
             "docfile": "ReadMe.txt",
-            "version": "0.4.0"
+            "version": "@MADLIB_PGXN_VERSION_STR@"
```
Comment (haradh1) [inline deploy/PGXN/META.json.in]: It should be ok as it stands, but I wonder why you didn't make this line major.minor.patch

Comment (fschopp) [inline deploy/PGXN/META.json.in]: OK. But the "version" in the outer block you would leave at @MADLIB_PGXN_VERSION_STR@?

**Comment by theory** [comment]
Awesome!

**Pull request merged**